src/main/java/teammates/ui/output/FeedbackSessionData.java

```diff
@@ -241,6 +271,36 @@ public void setPrivileges(InstructorPermissionSet privileges) {
         this.privileges = privileges;
     }
 
+    public void setStudentDeadlines(Map<String, Long> studentDeadlines) {
+        this.studentDeadlines = studentDeadlines;
+    }
+
+    public void setInstructorDeadlines(Map<String, Long> instructorDeadlines) {
+        this.instructorDeadlines = instructorDeadlines;
+    }
+
+    /**
+     * Filter out all the deadlines that are not for the given student.
+     */
+    public void filterDeadlinesForStudent(String studentEmailAddress) {
+        setStudentDeadlines(studentDeadlines.entrySet()
+                .stream()
+                .filter(entry -> entry.getKey().equals(studentEmailAddress))
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue)));
+        setInstructorDeadlines(new HashMap<>());
+    }
+
+    /**
+     * Filter out all the deadlines that are not for the given instructor.
+     */
+    public void filterDeadlinesForInstructor(String instructorEmailAddress) {
+        setStudentDeadlines(new HashMap<>());
+        setInstructorDeadlines(instructorDeadlines.entrySet()
+                .stream()
+                .filter(entry -> entry.getKey().equals(instructorEmailAddress))
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue)));
+    }
```

Comment:
>we seem to be finding all kv pairs that have the same key

Yea, I think the problem is because I think there is no way to get just that one item while using streams, but I think the root problem might just be the fact that I use streams in the first place. I'll do it without streams, which should make it clearer.

Reply:
It is merely a semantics problem. Just don't read that nice despite shorter lines.